### PR TITLE
SQ-58/Speaker details screen

### PR DESCRIPTION
**Pull request by rock3r**
Yet another screen done! As usual it's a big one (sorry) but there's a lot of models/fixtures that are ezpz to deal with. I realised we needed those as we moved along, otherwise that would've been a separate PR.

Difference with spec: for now we only show one speaker at a time, no viewpager.

speaker-details

### CODE REVIEW

File: app/src/main/java/net/squanchy/navigation/Navigator.java [hunk 1]
```diff
@@ -16,21 +16,38 @@ public Navigator(Context context) {
...
+        if (canResolve(intent)) {
+            context.startActivity(intent);
+        } else {
+            toExternalUrl("https://twitter.com/" + username);
+            context.startActivity(intent);
+        }
+    }
+
+    private boolean canResolve(Intent intent) {
+        return context.getPackageManager()
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/navigation/Navigator.java]: missing `!` (or so you said HA)

File: app/src/main/java/net/squanchy/service/repository/SpeakerRepository.java [hunk 2]
```diff
@@ -27,7 +28,30 @@ public SpeakerRepository(FirebaseDbService dbService, Checksum checksum) {
...
+                        Optional.fromNullable(firebaseSpeaker.twitter_username)
                 )));
     }
+
+    public Observable<Speaker> speaker(String speakerId) {
+        return dbService.speakers()
+                .map(firebaseSpeakers -> firebaseSpeakers.speakers)
+                .flatMap(Observable::fromIterable)
+                .filter(firebaseSpeaker -> firebaseSpeaker.id.equals(speakerId))
+                .map(firebaseSpeaker -> Speaker.create(
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/service/repository/SpeakerRepository.java]: extract this and the previous one (creating a speaker from `firebaseSpeaker`) to a method

File: app/src/main/java/net/squanchy/speaker/widget/SpeakerHeaderView.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        updatePhoto(speaker.photoUrl());
+
+        nameView.setText(speaker.name());
+
+        Optional<String> companyName = speaker.companyName();
+        if (companyName.isPresent()) {
+            companyView.setText(companyName.get());
+            companyView.setVisibility(VISIBLE);
+            // TODO support navigating to company website
+        }
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/speaker/widget/SpeakerHeaderView.java]: `else` and put `companyView` as `GONE` in case

**Review by fourlastor — CHANGES_REQUESTED** [review]
🍌 Joe is happy but there are a few comments to address

File: app/src/main/java/net/squanchy/speaker/widget/SpeakerHeaderView.java (hunk 3 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/speaker/widget/SpeakerHeaderView.java]: 259cc94c10d984ed85532061e89f4c6c80b4a765

File: app/src/main/java/net/squanchy/service/repository/SpeakerRepository.java (hunk 2 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/service/repository/SpeakerRepository.java]: c9e6999491047d6ffb17468b24dde5cb71b0aa83

File: app/src/main/java/net/squanchy/navigation/Navigator.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/navigation/Navigator.java]: 13cf9cfd9d18f60d2e436d34e053243c3094e1a1

**Review by fourlastor — APPROVED** [review]

**Pull request merged**